pilot/pkg/bootstrap/configcontroller.go

```diff
@@ -131,6 +131,14 @@ func (s *Server) initConfigController(args *PilotArgs) error {
 		})
 	}
 
+	if features.EnableFederation {
+		if s.federation == nil {
+			log.Errorf("Federation support disabled: federation not initialized")
```

Comment:
This log does not make sense, because it says "Federation support disabled" when `features.EnableFederation` is true.

The second thing is that we don't have to check if both conditions `features.EnabledFederation` and `s.federation == nil` are true, because they cannot happen together. The reason of that is the initialization of istiod server is interrupted when federation is enabled, but not succeeded.

I would like to simplify this condition to just:
```
if s.federation == nil {
    s.ConfigStores = append(s.ConfigStores, s.federation.ConfigStore())
}
```